src/core/memory.lua

```diff
@@ -69,15 +69,15 @@ function reserve_new_page ()
    -- Check that we have permission
    lib.root_check("error: must run as root to allocate memory for DMA")
    -- Is the kernel shm limit too low for huge pages?
-   if huge_page_size > tonumber(syscall.sysctl("kernel.shmmax")) then
+   if huge_page_size > tonumber(lib.firstline("/proc/sys/kernel/shmmax")) then
       -- Yes: fix that
-      local old = syscall.sysctl("kernel.shmmax", tostring(huge_page_size))
+      local old = lib.writefile("/proc/sys/kernel/shmmax", tostring(huge_page_size))
       io.write("[memory: Enabling huge pages for shm: ",
                "sysctl kernel.shmmax ", old, " -> ", huge_page_size, "]\n")
    else
-      -- No: try provisioning an additional page
-      local have = tonumber(syscall.sysctl("vm.nr_hugepages"))
+      local have = tonumber(lib.firstline("/proc/sys/vm/nr_hugepages"))
       local want = have + 1
+      lib.writefile("/proc/sys/vm/nr_hugepages", tostring(want))
       syscall.sysctl("vm.nr_hugepages", tostring(want))
```

Comment:
Yes please, if the fix is in the original PR we can merge it however we like and at least be sure that you ACKed it.